### PR TITLE
Add text cls python serv

### PR DESCRIPTION
**Pull request by applenob**
# Pull Request Summary

- 1. Add text classification serving,  helping infer or debug the `SavedModel` in python.
- 2. Fix the shape function for custom op `SentenceToIds`, which cause crash when inferring a `SavedModel` containing `JiebaCut` and `HAN`.
- 3. Update the unit test for `text_cls_task`.

# Test Plan
tested.

### CODE REVIEW

File: delta/layers/ops/kernels/jieba_op_test.py [hunk 1]
```diff
@@ -134,12 +135,13 @@ def test_jieba_cut_op_no_file(self):
...
                                     {sentence_in: ["吉林省长春药店"]})
         self.assertEqual("吉林省 长春 药店", sentence_out_res[0].decode("utf-8"))
-        sentence_out_res = test_one(sess, sentence_out,
+        sentence_out_res, shape_res = test_one(sess, [sentence_out, shape_op],
                                     {sentence_in: ["吉林省长春药店", "南京市长江大桥"]})
         self.assertEqual(
             "吉林省 长春 药店\n南京市 长江大桥",
             "\n".join([one_sen.decode("utf-8") for one_sen in sentence_out_res
                       ]))
+        logging.info(f"shape: {shape_res}")
```
Comment (zh794390558) [inline delta/layers/ops/kernels/jieba_op_test.py]: assert shape

File: delta/serving/base_frozen_model.py [hunk 2]
```diff
@@ -99,6 +99,24 @@ def init_session(self, model, gpu_str):
       self._graph = utils.load_frozen_graph(frozen_graph)
       self._sess = tf.Session(graph=self._graph, config=config)
 
+  def inspect_ops(self):
+    for op in self._graph.get_operations():
+      logging.info(op.name)
+
+  def debug(self):
+    feed_dict = self.get_test_feed_dict()
```
Comment (zh794390558) [inline delta/serving/base_frozen_model.py]: self.get_test_feed_dict() this not common for all fronzen_model class.
should be self.feed_dict().

File: delta/layers/ops/kernels/x_ops.cc [hunk 3]
```diff
@@ -236,6 +242,11 @@ Status NgramShapeFn(InferenceContext* c) {
 
 Status SentenceToIdsShapeFn(InferenceContext* c) {
   ShapeHandle sent = c->input(0);
+  // If rank unknown, return unknown shape.
+  if (!c->RankKnown(sent)) {
+    c->set_output(0, c->UnknownShape());
```
Comment (zh794390558) [inline delta/layers/ops/kernels/x_ops.cc]: need two output shape.

File: delta/layers/ops/kernels/x_ops.cc [hunk 4]
```diff
@@ -200,6 +200,12 @@ Status NgramShapeFn(InferenceContext* c) {
 
   ShapeHandle sent = c->input(0);
   int32 rank = c->Rank(sent);
+  // If rank unknown, return unknown shape.
+  if (!c->RankKnown(sent)) {
+    c->set_output(0, c->UnknownShape());
```
Comment (zh794390558) [inline delta/layers/ops/kernels/x_ops.cc]: two outputs. pls add more info about output.

File: delta/layers/ops/kernels/x_ops.cc [hunk 5]
```diff
@@ -200,6 +200,12 @@ Status NgramShapeFn(InferenceContext* c) {
 
   ShapeHandle sent = c->input(0);
   int32 rank = c->Rank(sent);
+  // If rank unknown, return unknown shape.
+  if (!c->RankKnown(sent)) {
```
Comment (zh794390558) [inline delta/layers/ops/kernels/x_ops.cc]: we should know when the input rank is unknown.

File: delta/layers/ops/kernels/jieba_op_test.py [hunk 6]
```diff
@@ -33,7 +33,7 @@ def test_one(sess, ops, inputs):
   t2 = time.time()
   logging.info("inputs: {}".format(inputs))
   logging.info("time cost: {}".format(t2 - t1))
-  logging.info("\n".join([one_sen.decode("utf-8") for one_sen in sentence_out]))
+  # logging.info("\n".join([one_sen.decode("utf-8") for one_sen in sentence_out]))
```
Comment (zh794390558) [inline delta/layers/ops/kernels/jieba_op_test.py]: logging.debug

File: delta/serving/eval_text_cls_pb.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+''' Frozen text classification model Evaluater'''
+import os
+import sys
+import tensorflow as tf
```
Comment (zh794390558) [inline delta/serving/eval_text_cls_pb.py]: import delta.compat as tf

**Comment by hankun11** [comment]
@applenob  

"tested" is not a test plan. Next time, you need to provide HOW to test your code rather than wether you have tested the code. Thank you!

image

**Pull request merged**